gnat2goto/driver/range_check.adb

```diff
@@ -144,20 +160,53 @@ package body Range_Check is
       end case;
    end Get_Bound_Of_Bounded_Type;
 
-   ----------------------------
-   -- Make_Range_Assert_Expr --
-   ----------------------------
+
+   function Make_Overflow_Assert_Expr (N : Node_Id; Value : Irep) return Irep
+   is
+      Value_Type : constant Irep :=
+        Follow_Symbol_Type (Get_Type (Value), Global_Symbol_Table);
+      Low_Value : constant Irep := Get_Bound (N, Value_Type, Bound_Low);
+      High_Value : constant Irep := Get_Bound (N, Value_Type, Bound_High);
+      Maybe_Casted_Value : constant Irep := Value;
```

Comment:
I need to cast `value` to the type of `low_value`. But if I casted the whole expression it would be evaluated in the wider type during the assertion function call. That's why I cast the left-right hand side of the binary expression separately. I will add a case for unary expressions (didn't think they can overflow).

Reply:
Unary negate can overflow...